bin/build_cmd.ml

```diff
@@ -47,7 +47,48 @@ let run_build_system ~common ~request =
       Cached_digest.invalidate_cached_timestamps ();
       let* setup = Import.Main.setup () in
       let request =
-        Action_builder.bind (Action_builder.of_memo setup) ~f:(fun setup -> request setup)
+        let open Action_builder.O in
```

Comment:
This implementation should be factored more cleanly. We'd like the experimental code not to overlap with stable code.